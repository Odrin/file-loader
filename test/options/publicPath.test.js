import webpack from '../helpers/compiler';
import { loaders } from '../helpers/stats';

describe('publicPath', () => {
  test('- {String}', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          publicPath: '/test/test',
        },
      },
    };

    const stats = await webpack('./file.js', config);
    const { err, src, map, meta } = loaders(stats, 1);

    expect(err).toMatchSnapshot();
    expect(src).toMatchSnapshot();
    expect(map).toMatchSnapshot();
    expect(meta).toMatchSnapshot();
  });

  test('- {Function}', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          publicPath(url) {
            return `test/${url}`;
          },
        },
      },
    };

    const stats = await webpack('./file.js', config);
    const { err, src, map, meta } = loaders(stats, 1);

    expect(err).toMatchSnapshot();
    expect(src).toMatchSnapshot();
    expect(map).toMatchSnapshot();
    expect(meta).toMatchSnapshot();
  });
});
