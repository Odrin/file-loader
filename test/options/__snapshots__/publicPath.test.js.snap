// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publicPath - {Function} 1`] = `Array []`;

exports[`publicPath - {Function} 2`] = `"module.exports = \\"test/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;

exports[`publicPath - {Function} 3`] = `undefined`;

exports[`publicPath - {Function} 4`] = `Object {}`;

exports[`publicPath - {String} 1`] = `Array []`;

exports[`publicPath - {String} 2`] = `"module.exports = \\"/test/test9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;

exports[`publicPath - {String} 3`] = `undefined`;

exports[`publicPath - {String} 4`] = `Object {}`;
